email_helper.user.js

```diff
@@ -0,0 +1,123 @@
+
+function extractEmailText(emailId) {
+    var emailBody = $('#email-body-' + emailId);
+    var emailHeader = $('#email-' + emailId);
+    if (emailBody.length == 0) { // This is the MAIN/currently opened email in conversation
+        emailBody = $(emailHeader).next();
+    }
+    var emailComponents = [];
+    var emailBodyAuthorComponents = emailBody.children('.comment-author');
+    emailComponents.push(emailHeader.children('.comment-date')[0].innerHTML);
+    emailComponents.push(emailHeader.children('.comment-title')[0].innerHTML);
+    for (var i = 0; i < emailBodyAuthorComponents.length; i++){
+        emailComponents.push(emailBodyAuthorComponents[i].innerHTML);
+    }
+    emailComponents.push('<p/>'); //extra line break between headers and body
+    emailComponents.push(emailBody.children('.comment-body')[0].innerHTML);
+
+    var text = '';
+    for (var j = 0; j < emailComponents.length; j++){
+        text += turndownService.turndown(emailComponents[j]) + '  \r\n';
+    }
+    return text;
+}
+
+function addEmailToLS(emailId) {
+    var emails = {};
+    if (localStorage.hasOwnProperty(localStorageKey)) {
+        emails = JSON.parse(localStorage[localStorageKey]);
+    }
+    emails[emailId] = extractEmailText(emailId);
+    localStorage.setItem(localStorageKey, JSON.stringify(emails));
+}
+
+function removeEmailFromLS(emailId) {
+    if (localStorage.hasOwnProperty(localStorageKey)) {
+        var emails = JSON.parse(localStorage[localStorageKey]);
+        delete emails[emailId];
+        localStorage.setItem(localStorageKey, JSON.stringify(emails));
+    }
+}
+
+function isEmailForwarded(emailId) {
+    if (localStorage.hasOwnProperty(localStorageKey)) {
+        var emails = JSON.parse(localStorage[localStorageKey]);
+        return emails.hasOwnProperty(emailId);
+    }
+    return false;
+}
+
+function clearLS() {
+    localStorage.setItem(localStorageKey, JSON.stringify({}));
+}
+
+function initForwardButton(emailId, forwardStatus){
+    var emailBody = $('#email-body-' + emailId);
+    if (emailBody.length == 0) { // This is the MAIN/currently opened email in conversation
+        emailBody = $('#email-' + emailId).next();
+    }
+    var emailButtons = emailBody.children('.reply-actions');
+    var forwardButton = emailButtons.find('.btn-forward');
+    if (!forwardButton.length) {
+        emailButtons.prepend('<a href="#" class="btn-control normal btn-forward">' + (forwardStatus ? '&#9745;' : '&#9744;') + ' Forward</a>');
+        forwardButton = emailButtons.find('.btn-forward');
+        forwardButton.on('click', function(evt){
+            evt.preventDefault();
+            if (isEmailForwarded(emailId)){
+                removeEmailFromLS(emailId);
+                evt.target.innerHTML = '&#9744;' + ' Forward';
+            } else {
+                addEmailToLS(emailId);
+                evt.target.innerHTML = '&#9745;' + ' Forward';
+            }
+        })
+    } //TODO: 'else' block?
+}
+
+(function() {
```

Comment:
I think a [`$(document).ready( ... )` wrapper](https://learn.jquery.com/using-jquery-core/document-ready/) would be more robust here. Maybe tampermonkey takes care of running the JS only when the DOM is fully loaded, but I'm pretty sure greasemonkey does not.

Reply:
This is the wrapper that TamperMonkey provides within default userscript template, so I just used it.

Reply:
Changed in https://github.com/taustation-fan/userscripts/pull/32/commits/65ef0ca2d9ac5ec3c3bd2ee2da08e780c86d8d4c